packages/reakit/src/Composite/CompositeState.ts

```diff
@@ -259,12 +259,15 @@ function reducer(
         );
         return reducer(state, { ...action, type: "move", id: nextItem?.id });
       }
-      const oppositeOrientation = getOppositeOrientation(
-        // If it's a grid and orientation is not set, it's a next/previous
-        // call, which is inherently horizontal. up/down will call next with
-        // orientation set to vertical by default (see below on up/down cases).
-        isGrid ? orientation || "horizontal" : orientation
-      );
+      const oppositeOrientation =
+        orientation === "both"
+          ? "!both"
```

Comment:
I believe is better to add "both" to the `getOpositeOrientation` map:
```ts
const map = {
  horizontal: "vertical",
  vertical: "horizontal",
  both: "both"
} as const;
```
In this way, we won't need the [`Exclude<>` below](https://github.com/reakit/reakit/pull/816/files#diff-ed4f881f28fdb2988f5e1fcf07b02c37b04ac245f224c68c71e43543afc16f02), and we make sure `getOppositeOrientation` can handle the `"both"` case 👍🏼